### DISCUSSION TITLE
Suddenly broadcast of teacher's screen to students

### DISCUSSION THREAD

**Discussion by rweilert** [epoptes/epoptes#121] [Q&A]
Yesterday we saw a phenomenon which we cannot explain. Our setup is Ubuntu 18.04 with Mate and LTSP5. There is one server for everyone, teacher has Epoptes, and I have Epoptes (as I am the admin), we both running as thin clients like everyone. I can see everyone, the teacher only sees what is not in X-Hidden. As I reported earlier, since our last update, I can see myself now.

The teacher had Epoptes NOT started yet, only looked up our classbook app, clicking here and there. But I HAD Epoptes running just to see how many people were logged in, but at that time I had it on another screen and was doing something else.

Now, teacher came up and said that suddenly her screen appeared on all students' screens (the classbook) and covered their program they were working with, but only in the classroom. Nobody else in the house saw this, I didn't see it either (there is another lab, another 4 clients in the isle, and clients in the classrooms). Maybe the other lab and the 4 clients weren't used then, and the other clients are in X-Hidden anyway.

When the students told her, she first started Epoptes, but she couldn't stop the broadcast or didn't know how. Then she shutdown her client, and students' screens were freed.

Idea 1: is there some service running for broadcasts even when Epoptes is off? And if so, is there a special key combination or something like that which would start it?

Idea 2: If not, might my own running Epoptes instance have something to do with it? Maybe it lets run some kind of service in the background for everyone else with Epoptes access?

Idea 3: Might this be a side effect from "now I can see myself"?

I would like to avoid such things in the future.

Regards
Rolf

**Comment by alkisg** [comment]
Hi Rolf, no, epoptes doesn't have any hidden services and no key combinations for this and no way to broadcast oneself when the GUI is closed.
Regarding "now I can see myself", no updates were done to epoptes since 2020-01-24.

**Comment by rweilert** [comment]
Well, good to know - and yes, we only recently updated (as I wrote), and so I have this only now. 

So what might have caused this?

**Comment by alkisg** [comment]
If you weren't clicking around in the Epoptes GUI and you didn't broadcast her screen yourself, no idea, it doesn't sound doable from the Epoptes code base.
Broadcast are even password protected, so it's not possible to reuse the same broadcast for other clients, much less after reboots.

**Comment by alkisg** [comment]
The only thing that sounds remotely related, is that LTSP5 thin client sessions do not terminate properly, so maybe it's related to an old broadcast of her that was still running on the server and that was never terminated, and somehow her connecting again made DISPLAY relevant again.

There are some discussions about properly ending LTSP5 thin client session processes in the mailing list, they should be searchable in sourceforge.
A keyword for searching: KillUserProcesses=yes in /etc/systemd/logind.conf

**Comment by alkisg** [comment]
See also:
https://github.com/ltsp/community/issues/92#issuecomment-578800253
https://github.com/mate-desktop/mate-session-manager/issues/165
https://sourceforge.net/p/ltsp/mailman/message/37039484/

**Comment by rweilert** [comment]
Thanks a lot, that could be an explanation. We sometimes have had problems with such old processes.

I'll see if I can do something about it.

Regards
Rolf

**Comment by rweilert** [comment]
There were indeed a number of processes (of the teacher) running even after some days. 

So I now set KillUserProcesses=yes in /etc/systemd/logind.conf and we'll see what happens. Do I have to do anything more than just wait for the next login/logout cycle? Do I have to reboot the server? Or do I have to make a new image for it?

Regards
Rolf
